src/main.rs

```diff
@@ -103,23 +120,34 @@ async fn new(cookies: &CookieJar<'_>, request: Json<NewRequest>) -> Result<Strin
 			round_pubkey_bytes,
 		)
 		.map_err(|_| Status::InternalServerError)?;
+		
 		// 3. add to storage
-		store::write(mmr_store.clone());
-		// sign and send the call
-		let from = dev::alice();
-		let _events = client
-			.tx()
-			.sign_and_submit_then_watch_default(&call, &from)
-			.await
-			.map_err(|_| Status::InternalServerError)?;
-		Ok("MMR proxy account creation successful!".to_string())
+		let username_string: String = username.into();
+		let insert_result = db.write(username_string, mmr_store.clone(), None).await;
+		match insert_result {
+			Err(_e) => Err(Status::InternalServerError),
+			Ok(_result) => {
+				// sign and send the call
+				let from = dev::alice();
+				let _events = client
+					.tx()
+					.sign_and_submit_then_watch_default(&call, &from)
+					.await
+					.map_err(|_| Status::InternalServerError)?;
+				Ok("MMR proxy account creation successful!".to_string())
+			}
+		}
```

Comment:
It is best practice to try to avoid conditionals when possible, something like this would be easier to read:
```suggestion
		db.write(username_string, mmr_store.clone(), None)
			.await
			.map_err(|_| Status::InternalServerError)?;

		// sign and send the call
		let from = dev::alice();
		let _events = client
			.tx()
			.sign_and_submit_then_watch_default(&call, &from)
			.await
			.map_err(|_| Status::InternalServerError)?;
		Ok("MMR proxy account creation successful!".to_string())
```